### PR TITLE
Add support for passing authentication credentials while creating a connection

### PR DESCRIPTION
**Pull request by sumeetkoli**
This change will allow clients to use connect to a mongo instance which has authentication enabled by passing vaild credentials.

### CODE REVIEW

File: src/mongrove/core.clj [hunk 1]
```diff
@@ -178,6 +181,8 @@
   (let [opts (merge default-opts opts)
         {:keys [read-preference read-concern write-concern
                 retry-reads retry-writes]} opts
+        credential (when (some? (and username password))
```
Comment (ghoseb) [inline src/mongrove/core.clj]: I think the `some?` is redundant. Can you verify that we really need this?

Comment (ghoseb) [reply src/mongrove/core.clj]: Please squash this commit.

**Review by ghoseb — CHANGES_REQUESTED** [review]
Please squash one commit and review your use of `some?`.

File: src/mongrove/core.clj (hunk 1 above)
Comment (sumeetkoli) [reply src/mongrove/core.clj]: Thank you for the review. This is done.

**Review by ghoseb — APPROVED** [review]
Looks good to me. 👍🏽

**Pull request merged**